File: Page/src/app/models/DTO/noticiaDTO.ts
import { EquipoModel } from "../equipo-model";

export class NoticiaDTO {
  usuarioId: number;
  imagen: string;
  html: string;
  fecha: Date;
  titulo: string;
  equipos?: EquipoModel[];
  torneos?: EquipoModel[];
  jugadores?: EquipoModel[];
  constructor( usuario: number, imagen: string, html: string, fecha: Date, titulo: string, equipos?: EquipoModel[], torneos?: EquipoModel[], jugadores?: EquipoModel[]) {
    this.usuarioId = usuario;
    this.imagen = imagen;
    this.html = html;
    this.fecha = fecha;
    this.titulo = titulo;
    this.equipos = equipos;
    this.torneos = torneos;
    this.jugadores = jugadores;
  }



  getImagen(): string {
    return this.imagen;
  }

  setImagen(imagen: string): void {
    this.imagen = imagen;
  }

  getHtml(): string {
    return this.html;
  }

  setHtml(html: string): void {
    this.html = html;
  }

  getFecha(): Date {
    return this.fecha;
  }

  setFecha(fecha: Date): void {
    this.fecha = fecha;
  }

  getTitulo(): string {
    return this.titulo;
  }

  setTitulo(titulo: string): void {
    this.titulo = titulo;
  }
}

File: Api/src/main/java/com/cstv/cstv/domain/NoticiaDTO.java
package com.cstv.cstv.domain;

import java.sql.Date;
import java.util.List;

public class NoticiaDTO {
    private Long id;
    private Long usuarioId;
    private String imagen;
    private String html;
    private Date fecha;
    private String titulo;
    private List<Long> equiposIds;
    private List<Long> torneosIds;
    private List<Long> jugadoresIds;
    // Getters y Setters

    public List<Long> getEquiposIds() {
        return equiposIds;
    }

    public void setEquiposIds(List<Long> equiposIds) {
        this.equiposIds = equiposIds;
    }

    public List<Long> getTorneosIds() {
        return torneosIds;
    }

    public void setTorneosIds(List<Long> torneosIds) {
        this.torneosIds = torneosIds;
    }

    public List<Long> getJugadoresIds() {
        return jugadoresIds;
    }

    public void setJugadoresIds(List<Long> jugadoresIds) {
        this.jugadoresIds = jugadoresIds;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getUsuarioId() {
        return usuarioId;
    }

    public void setUsuarioId(Long usuarioId) {
        this.usuarioId = usuarioId;
    }

    public String getImagen() {
        return imagen;
    }

    public void setImagen(String imagen) {
        this.imagen = imagen;
    }

    public String getHtml() {
        return html;
    }

    public void setHtml(String html) {
        this.html = html;
    }

    public Date getFecha() {
        return fecha;
    }

    public void setFecha(Date fecha) {
        this.fecha = fecha;
    }

    public String getTitulo() {
        return titulo;
    }

    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }
}
